Solaris specific changes needed for how we configure curl.

These changes need to be sent upstream, but before that can be done,
they need to be reworked. They also need to apply to configure.ac,
not configure.

--- configure.~1~	2016-07-21 12:16:35.000000000 +0300
+++ configure	2016-08-08 23:13:26.720777620 +0300
@@ -3083,7 +3083,7 @@
 
 
 
-ac_config_headers="$ac_config_headers lib/curl_config.h include/curl/curlbuild.h"
+ac_config_headers="$ac_config_headers lib/curl_config.h"
 
 
 
@@ -3678,12 +3678,6 @@
 fi
 
 
-
-
-if test -f ${srcdir}/include/curl/curlbuild.h; then
-  rm -f ${srcdir}/include/curl/curlbuild.h
-fi
-
 CURLVERSION=`$SED -ne 's/^#define LIBCURL_VERSION "\(.*\)"/\1/p' ${srcdir}/include/curl/curlver.h`
 
   xc_prog_cc_prev_IFS=$IFS
@@ -21798,7 +21792,7 @@
               gss_libname="gss"
               ;;
            *)
-              gss_libname="gssapi"
+              gss_libname="gss"
               ;;
            esac
 
@@ -21818,7 +21812,7 @@
         LIBS="-lgss $LIBS"
         ;;
      *)
-        LIBS="-lgssapi $LIBS"
+        LIBS="-lgss $LIBS"
         ;;
      esac
   fi
@@ -21949,7 +21943,7 @@
         if test x$cross_compiling != xyes; then
             PKGTEST="yes"
     fi
-    PREFIX_OPENSSL=/usr/local/ssl
+    PREFIX_OPENSSL="/usr"
     LIB_OPENSSL="$PREFIX_OPENSSL/lib$libsuff"
     ;;
   off)
@@ -28347,7 +28341,7 @@
     #
   { $as_echo "$as_me:${as_lineno-$LINENO}: checking for curl_socklen_t data type" >&5
 $as_echo_n "checking for curl_socklen_t data type... " >&6; }
-  curl_typeof_curl_socklen_t="unknown"
+  curl_typeof_curl_socklen_t="socklen_t"
   for arg1 in int SOCKET; do
     for arg2 in 'struct sockaddr' void; do
       for t in socklen_t int size_t 'unsigned int' long 'unsigned long' void; do
@@ -41860,7 +41854,6 @@
 do
   case $ac_config_target in
     "lib/curl_config.h") CONFIG_HEADERS="$CONFIG_HEADERS lib/curl_config.h" ;;
-    "include/curl/curlbuild.h") CONFIG_HEADERS="$CONFIG_HEADERS include/curl/curlbuild.h" ;;
     "depfiles") CONFIG_COMMANDS="$CONFIG_COMMANDS depfiles" ;;
     "libtool") CONFIG_COMMANDS="$CONFIG_COMMANDS libtool" ;;
     "Makefile") CONFIG_FILES="$CONFIG_FILES Makefile" ;;
